content/blog/managing-aws-credentials-on-cicd/index.md

```diff
@@ -0,0 +1,231 @@
+1. Change the state of the previous access key to inactive.
+1. Validate that your applications are still working as expected.
+1. Delete the inactive access key.
+
+## During deployments, exchange the IAM user credentials for a more privileged IAM role
+
+Now that we have a regularly rotated, low-privileged credential to use during deployments, the next step is to use “assume role” to exchange those IAM User credentials for an IAM Role credential. This sounds confusing at first. Why exchange one credential for another? Why not just use the same credentials? In other words, what is an IAM Role, and why is this so much better than an IAM User? The credentials used to assume an IAM Role are only valid for a short period (minutes to hours). After a fixed duration, there is no threat if disclosed.
+
+Earlier, we mitigate potentially exposing IAM User credentials by using our CI/CD provider’s ability to store secret values and automatically rotating credentials. Using the IAM Role credentials adds another layer of security, but within the context of the machine itself during the deployment.
+
+So how do you use these credentials? The AWS Secure Token Service can do this, but in practice, you’ll want to use a lightweight tool to do this.
+
+### Configuring AWS Security Group
+
+The following Pulumi snippet creates the AWS security group.
+
+```ts
+security_group = aws.ec2.SecurityGroup('secgrp',
+    description='Enable HTTP access',
+    ingress=[
+        { 'protocol': 'tcp', 'from_port': 22, 'to_port': 22, 'cidr_blocks': ['0.0.0.0/0'] },
+        { 'protocol': 'tcp', 'from_port': 8080, 'to_port': 8080, 'cidr_blocks': ['0.0.0.0/0'] }
+    ],
+    egress=[
+        { 'protocol': '-1', 'from_port': 0, 'to_port': 0, 'cidr_blocks': ['0.0.0.0/0'] }
+        ]
+)
+```
+
+### Auditing Access
+
+Now that we have secure access, we need to audit it too.
+
+[see auditing blog]
+
+### What about your Pulumi Credentials?
+
+So far, we’ve written extensively about securing AWS access tokens, but aren’t Pulumi credentials sensitive as well? And wouldn’t they need to be secure?
+
+Yes. Though as much as we like to think our product provides information that eases deployments, understanding the history, state, and topology of your cloud resources is not as important as having access to your cloud resources themselves.
+
+You’ll want to follow the same approach we described for AWS credentials with your Pulumi Access tokens. We support revoking those credentials in the Pulumi UI, and you'll want to recreate and rotate them regularly as well.
+
+It’s best practice to store them using AWS SecretsManager to [load and store those credentials](https://aws.amazon.com/secrets-manager/).
+
+Here is an example for reading/writing a Pulumi Access Token from AWS Secrets Manager. This, of course, assumes that the current AWS credentials (e.g., the result of assume-role) have access. Create an access token using the Pulumi console and copy it to a file.
```

Comment:
Maybe call out that you need to have the IAM User assume role into the more privileged IAM Role account in order to read the secret from secrets manager? (Since it isn't obviously "who" should have access to this secret.)